santoku/utils/test_configuration_manager.py

```diff
@@ -0,0 +1,152 @@
+import pytest
+
+from santoku.utils.configuration_manager import (
+    ConfigurationManager,
+    UndefinedConfiguration,
+    UndefinedSetting,
+)
+
+
+@pytest.fixture(scope="class")
+def boolean_setting():
+    return "BOOLEAN_SETTING"
+
+
+@pytest.fixture(scope="class")
+def string_setting():
+    return "STRING_SETTING"
+
+
+@pytest.fixture(scope="function")
+def default_configuration(boolean_setting, string_setting):
+    return {
+        "NAME": "DEFAULT",
+        "CONFIG": {boolean_setting: True, string_setting: "default_string_value"},
+    }
+
+
+@pytest.fixture(scope="function")
+def test_configuration(boolean_setting, string_setting):
+    return {"NAME": "TEST", "CONFIG": {boolean_setting: False, string_setting: "test_string_value"}}
+
+
+@pytest.fixture(scope="function")
+def configuration_manager(default_configuration, test_configuration):
+    predefined_configurations = [default_configuration, test_configuration]
+    return ConfigurationManager(
+        predefined_configurations=predefined_configurations,
+        default_configuration=default_configuration["NAME"],
+    )
+
+
+class TestConfigurationManager:
+    def test_get_setting(
+        self, boolean_setting, string_setting, default_configuration, configuration_manager
+    ):
+        # Get a setting from the current configuration. Success expected.
+        expected_value = default_configuration["CONFIG"][boolean_setting]
+        obtained_value = configuration_manager.get_setting(key=boolean_setting)
+        assert obtained_value == expected_value
+
+        expected_value = default_configuration["CONFIG"][string_setting]
+        obtained_value = configuration_manager.get_setting(key=string_setting)
+        assert obtained_value == expected_value
+
+        # Get an undefined setting from the current configuration. Failure expected.
+        undefined_setting_key = "undefined_setting"
+        expected_message = f"Setting '{undefined_setting_key}' undefined."
+        with pytest.raises(UndefinedSetting, match=expected_message) as e:
+            configuration_manager.get_setting(key=undefined_setting_key)
+
+    def test_list_settings(
+        self, boolean_setting, string_setting, default_configuration, configuration_manager
+    ):
+        # List all settings defined in the current configuration. Success expected.
+        expected_settings = [(boolean_setting, default_configuration["CONFIG"][boolean_setting])]
+        expected_settings.append((string_setting, default_configuration["CONFIG"][string_setting]))
+        obtained_settings = configuration_manager.list_settings()
+        assert set(obtained_settings) == set(expected_settings)
+
+    def test_set_setting(self, boolean_setting, default_configuration, configuration_manager):
+        # Change the value of an already defined setting in the current configuration.
+        # Success expected.
+        expected_value = default_configuration["CONFIG"][boolean_setting]
```

Comment:
I do not understand this test, there is no call to the function being tested, is there?

Also, to keep tests independent, maybe we should avoid using other functions in the test other than the one being tested. In this case, instead of using `get_setting`, we could go a bit lower level and access the internal state of the configuration_manager class itself (i.e. the `settings` dictionary)

Same thing goes for most of the test suite.